app/pages/lab-pages-editor/DataManager.jsx

```diff
@@ -0,0 +1,82 @@
+/*
+Data Manager
+Responsible for fetching, updating, and saving Workflow resources from/to
+the Panoptes service.
+ */
+
+import apiClient from 'panoptes-client/lib/api-client'
+import { useEffect, useState } from 'react'
+import { WorkflowContext } from './context.jsx'
+
+export default function DataManager ({
+  // key: to ensure DataManager renders FRESH (with states reset) whenever workflowId changes, use <DataManager key={workflowId} ... />
+  children,
+  workflowId,
+}) {
+  let initialised = false
+  const [ apiData, setApiData ] = useState({
+    workflow: null,
+    status: 'ready',
+  })
+
+  // Fetch workflow when the component loads for the first time.
+  // See notes about 'key' prop, to ensure states are reset:
+  // https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes
+  // Also see general pattern notes:
+  // https://react.dev/reference/react/useEffect#fetching-data-with-effects
+  useEffect(async function whenComponentMounts_doInitialise () {
+    console.log('+++ DataManager: fetchWorkflow', workflowId)
+    if (initialised) return 
+    if (!workflowId) return
+
+    try {
+      setApiData({
+        workflow: null,
+        status: 'fetching'
+      })
+
+      const wf = await apiClient.type('workflows').get(workflowId)
+      if (!wf) throw new Error('No workflow')
+
+      setApiData({
+        workflow: wf,
+        status: 'ready'
+      })
+      initialised = true
+
+    } catch (err) {
+      console.error('DataManager: ', err)
+      setApiData({
+        workflow: null,
+        status: 'error'
+      })
+      initialised = true  // Don't re-run the initialisation function again
+    }
+
+    return () => { initialised = false }
+  }, [ workflowId ])
```

Comment:
```suggestion
  }, [ initialised, workflowId ])
```

This hook also depends on `initialised`. All of an effect hook's reactive dependencies must be declared here. You can't choose which values a `useEffect` hook will react to.
https://react.dev/reference/react/useEffect#specifying-reactive-dependencies

Reply:
I've removed the `initialised` var altogether, to avoid additional complications at this stage.

My tests seem to confirm that the workflow fetch triggers only once, so that flag seems unnecessary. I'm not worried about a race condition on _fetching,_ but I will re-evaluate that [intialised/active/ignore](https://maxrozen.com/race-conditions-fetching-data-react-with-useeffect) variable when I'm working on _updating_ the workflow.

Reply:
You’ve keyed this component by workflow ID in the pages editor, so it’s impossible for `workflowId` to change from one render to another. Each new workflow creates a new component.

Reply:
If it’s useful, take a look at the IIIF editor to see how it separates data-fetching state from editing state. You’ll probably run into the same problems that I did, so I suspect the solutions will be the same.